pkg/importer/util.go

```diff
@@ -52,3 +54,28 @@ func StreamDataToFile(dataReader io.Reader, filePath string) error {
 	}
 	return nil
 }
+
+// StreamDataToBlockDevice provides a function to stream the specified io.Reader to the specified block device
+func StreamDataToBlockDevice(dataReader io.Reader, filePath string) error {
+	var fd int
+	var err error
+
+	fd, err = syscall.Open(filePath, os.O_CREATE|os.O_WRONLY, 0777)
+	defer syscall.Close(fd)
+	if err != nil {
+		return errors.Wrapf(err, "could not open device file %q", filePath)
+	}
+
+	buf := new(bytes.Buffer)
+	glog.V(1).Infof("reading from dataReader....\n")
+	buf.ReadFrom(dataReader)
+	glog.V(1).Infof("writing to block device....\n")
+	if _, err = syscall.Write(fd, buf.Bytes()); err != nil {
```

Comment:
You are reading the entire stream into memory and then writing the entire stream out.  This obviously does not scale.  Why can't you use `io.Copy` like `StreamDataToFile`?  It reads/writes in chunks.

Reply:
It would be trivial two write your own `Writer` that wraps calls to `syscall.Write`.  Or maybe `os.NewFile` will work?